src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -274,18 +274,17 @@ struct TreeTraversal<BVH, Predicates, Callback, NearestPredicateTag>
         heap(UnmanagedStaticVector<PairIndexDistance>(buffer.data(),
                                                       buffer.size()));
 
-    using PairNodePtrDistance = Kokkos::pair<Node const *, float>;
-    Stack<PairNodePtrDistance> stack;
-    // Do not bother computing the distance to the root node since it is
-    // immediately popped out of the stack and processed.
-    stack.emplace(bvh_.getRoot(), 0.);
+    Node const *stack_node[64];
+    float stack_dist[64];
```

Comment:
Did you try 
```C++
Kokkos::pair<Node const *, float> stack[64];
```

Reply:
Yes. It complained about not having the default constructor, or something like that.